src/indexed_reader.jl

```diff
@@ -13,6 +13,9 @@ struct IndexedBSONReader{R <: BSONReader} <: AbstractBSONReader
 end
 
 @inline Base.sizeof(reader::IndexedBSONReader) = sizeof(reader.reader)
+@inline function Base.foreach(f, reader::IndexedBSONReader)
+    foreach(f, Map(identity), reader.reader)
+end
```

Comment:
I think this should be sufficient, no?
```suggestion
@inline Base.foreach(f, reader::IndexedBSONReader) = foreach(f, reader.reader)
```